### PR TITLE
[CodeCompletion] Ensure all ExtensionDecl's extended types are computed

### PR DESCRIPTION
**Pull request by rintaro**
Fixes an assertion failure in ASTScope lookup

rdar://problem/67102794

### CODE REVIEW

**Comment by rintaro** [comment]
@swift-ci Please smoke test

File: lib/IDE/ExprContextAnalysis.cpp [hunk 1]
```diff
@@ -49,16 +49,20 @@ void swift::ide::typeCheckContextAt(DeclContext *DC, SourceLoc Loc) {
...
-    for (auto typeCtx = DC->getInnermostTypeContext(); typeCtx != nullptr;
-         typeCtx = typeCtx->getParent()->getInnermostTypeContext()) {
-      if (auto *ext = dyn_cast<ExtensionDecl>(typeCtx))
-        extensions.push_back(ext);
-    }
-    while (!extensions.empty()) {
-      extensions.back()->computeExtendedNominal();
-      extensions.pop_back();
-    }
+    class ExtensionBinder: public ASTWalker {
```
Comment (CodaFi) [inline lib/IDE/ExprContextAnalysis.cpp:52]: Why does this traverse the AST instead of iterating over the top level declarations in a file or something?

Comment (rintaro) [reply lib/IDE/ExprContextAnalysis.cpp:52]: I thought we need to handle something like:
```
struct Something {

extension A: B.#^HERE^# {
}
```
But I didn't know this already has `ErrorType` as the extended type.
Will fix it.

Comment (rintaro) [reply lib/IDE/ExprContextAnalysis.cpp:52]: > this already has ErrorType as the extended type.

Well, this is not true. But somehow 
```
struct Something {

extension A: B.#^HERE^# {
}
```
doesn't crash. I will investigate.

Comment (CodaFi) [reply lib/IDE/ExprContextAnalysis.cpp:52]: Malformed ASTs makes sense. Thanks.

Comment (rintaro) [reply lib/IDE/ExprContextAnalysis.cpp:52]: Ah, I see. Non-toplevel `extension` is `canNeverBeBound()`. So we don't need to compute the extended nominal unless it's top-level.

File: lib/IDE/ExprContextAnalysis.cpp [hunk 2]
```diff
@@ -46,9 +46,11 @@ void swift::ide::typeCheckContextAt(DeclContext *DC, SourceLoc Loc) {
   while (isa<AbstractClosureExpr>(DC))
     DC = DC->getParent();
 
-  // Make sure the extension has been bound, in case it is in an inactive #if
-  // or something weird like that.
```
Comment (rintaro) [inline lib/IDE/ExprContextAnalysis.cpp:50]: Code completion never happens inside inactive `#if` since #33475

**Comment by rintaro** [comment]
@swift-ci Please smoke test

**Review by nathawes — APPROVED** [review]
LGTM!

**Pull request merged**